llmfoundry/callbacks/loss_perp_v_len_callback.py

```diff
@@ -0,0 +1,349 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, Mapping, Optional, Tuple
+
+import torch
+from composer.core import Callback, State
+from composer.loggers import Logger, MLFlowLogger
+from composer.utils import dist
+from torchmetrics import Metric
+
+from llmfoundry.models.mpt import ComposerMPTCausalLM
+
+__all__ = [
+    'LossPerpVsContextLengthLogger',
+]
+
+
+class LossPerpVsContextLengthLogger(Callback):
+    """Logs the average loss and perplexity for every context length.
```

Comment:
Should we note that this only works with MLflow? (Is that true or am I misremembering something?)

Reply:
Done.